loopd/macaroons.go

```diff
@@ -150,8 +147,16 @@ func (d *Daemon) startMacaroonService() error {
 	var err error
 	d.macaroonService, err = macaroons.NewService(
 		d.cfg.DataDir, loopMacaroonLocation, false,
-		macDatabaseOpenTimeout, macaroons.IPLockChecker,
+		loopdb.DefaultLoopDBTimeout, macaroons.IPLockChecker,
 	)
+	if err == bbolt.ErrTimeout {
+		return fmt.Errorf("error while trying to open %s/%s: "+
```

Comment:
Great explanatory error!

Just wondering where these errors should live. `startMacaroonService` and `NewBoltSwapStore` are both called from our `initialize` function for the daemon. WDYT about bubbling the bbolt.ErrTimeout all the way up (or wrap it so that we can also have the db name), and then we can have a single descriptive error in the [daemon](https://github.com/lightninglabs/loop/blob/master/loopd/daemon.go#L130)?

If we do it like that, the terminal-specific instructions could also be handled in terminal by checking the response for `StartAsSubserver`'s [response](https://github.com/lightninglabs/loop/blob/master/loopd/daemon.go#L171).

Reply:
Okay, I've implemented your suggestion. What do you think?

Reply:
Looks great!